### ISSUE TITLE
Debian 8 arm linux gnueabi-hf

### ISSUE THREAD

**Issue by mickae1** [netdata/netdata#10076]
Hi,

I would like to find a package compatible with Debian 8 arm linux gnueabi-hf.

do you know where I can find it ?

**Comment by Ferroin** [comment]
We don’t currently provide packages for any architectures other than x86 and 64-bit x86, though we do have plans to support at least ARM (both 32 and 64-bit). Unfortunately said plans are currently on-hold pending the completion of a major rewrite of our binary packaging handling code.

Due to Debian 8 being in it’s 'extended long-term-support period' we are not currently providing packages for Debian 8 at all.

---

_However_, last I knew, Netdata itself still builds correctly on Debian 8, which means it should be possible to build a Debian package yourself for your desired platform just using the regular Debian package building process (just make sure to copy `contrib/debian` to `debian` in the Netdata source tree before trying to build the package).

**Comment by mickae1** [comment]
I have this error:

> 
> make[2]: Entering directory '/home/micka/netdata-1.26.0/externaldeps/libbpf/libbpf-0.0.9_netdata-1/src'
> Package libelf was not found in the pkg-config search path.
> Perhaps you should add the directory containing `libelf.pc'
> to the PKG_CONFIG_PATH environment variable
> No package 'libelf' found
> mkdir -p build//staticobjs
> cc -I. -I../include -I../include/uapi -DCOMPAT_NEED_REALLOCARRAY -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -D_FORTIFY_SOURCE=2 -c bpf.c -o build//staticobjs/bpf.o
> bpf.c:56:4: error: #error __NR_bpf not defined. libbpf does not support your arch.
>  #  error __NR_bpf not defined. libbpf does not support your arch.
>     ^
> bpf.c: In function 'sys_bpf':
> bpf.c:68:17: error: '__NR_bpf' undeclared (first use in this function)
>   return syscall(__NR_bpf, cmd, attr, size);
>                  ^
> bpf.c:68:17: note: each undeclared identifier is reported only once for each function it appears in
> Makefile:102: recipe for target 'build//staticobjs/bpf.o' failed
> make[2]: *** [build//staticobjs/bpf.o] Error 1
> make[2]: Leaving directory '/home/micka/netdata-1.26.0/externaldeps/libbpf/libbpf-0.0.9_netdata-1/src'
> debian/rules:37: recipe for target 'override_dh_auto_configure' failed
> make[1]: *** [override_dh_auto_configure] Error 1
> make[1]: Leaving directory '/home/micka/netdata-1.26.0'
> debian/rules:19: recipe for target 'binary' failed
>

**Comment by mickae1** [comment]
I don't have the rpm-tools:
-- Looking for RPMTools... - rpmbuild NOT FOUND
---------------------------------------------------------------------
  Settings:  (For more help do cmake -LH <srcpath>)
---------------------------------------------------------------------
 LWS_WITH_STATIC = ON
 LWS_WITH_SHARED = ON
 LWS_WITH_SSL = ON (SSL Support)
 LWS_SSL_CLIENT_USE_OS_CA_CERTS = 1
 LWS_WITH_WOLFSSL = OFF (wolfSSL/CyaSSL replacement for OpenSSL)
 LWS_WITH_MBEDTLS = OFF (mbedTLS replacement for OpenSSL)
 LWS_WITHOUT_BUILTIN_SHA1 = OFF
 LWS_WITHOUT_BUILTIN_GETIFADDRS = OFF
 LWS_WITHOUT_CLIENT = OFF
 LWS_WITHOUT_SERVER = OFF
 LWS_LINK_TESTAPPS_DYNAMIC = OFF
 LWS_WITHOUT_TESTAPPS = OFF
 LWS_WITHOUT_TEST_SERVER = OFF
 LWS_WITHOUT_TEST_SERVER_EXTPOLL = OFF
 LWS_WITHOUT_TEST_PING = OFF
 LWS_WITHOUT_TEST_CLIENT = OFF
 LWS_WITHOUT_EXTENSIONS = ON
 LWS_WITH_LATENCY = OFF
 LWS_WITHOUT_DAEMONIZE = ON
 LWS_WITH_LIBEV = OFF
 LWS_WITH_LIBUV = OFF
 LWS_WITH_LIBEVENT = OFF
 LWS_IPV6 = OFF
 LWS_UNIX_SOCK = OFF
 LWS_WITH_HTTP2 = 1
 LWS_SSL_SERVER_WITH_ECDH_CERT = OFF
 LWS_MAX_SMP = 1
 LWS_HAVE_PTHREAD_H = 1
 LWS_WITH_CGI = OFF
 LWS_HAVE_OPENSSL_ECDH_H = 1
 LWS_HAVE_SSL_CTX_set1_param = 1
 LWS_HAVE_RSA_SET0_KEY = 1
 LWS_WITH_HTTP_PROXY = OFF
 LIBHUBBUB_LIBRARIES =
 PLUGINS =
 LWS_WITH_ACCESS_LOG = OFF
 LWS_WITH_SERVER_STATUS = OFF
 LWS_WITH_LEJP = ON
 LWS_WITH_LEJP_CONF = ON
 LWS_WITH_SMTP = OFF
 LWS_WITH_GENERIC_SESSIONS = OFF
 LWS_STATIC_PIC = OFF
 LWS_WITH_RANGES = OFF
 LWS_PLAT_OPTEE = OFF
 LWS_WITH_ESP32 = OFF
 LWS_WITH_ZIP_FOPS = OFF
 LWS_AVOID_SIGPIPE_IGN = OFF
 LWS_WITH_STATS = OFF
 LWS_WITH_SOCKS5 = YES
 LWS_HAVE_SYS_CAPABILITY_H =
 LWS_HAVE_LIBCAP =
 LWS_WITH_PEER_LIMITS = OFF
 LWS_HAVE_ATOLL = 1
 LWS_HAVE__ATOI64 =
 LWS_HAVE_STAT32I64 =
 LWS_HAS_INTPTR_T = 1
 LWS_WITH_EXPORT_LWSTARGETS = ON
 LWS_WITH_ABSTRACT =
---------------------------------------------------------------------
-- Configuring done

**Comment by Ferroin** [comment]
Hmm, I forgot we have unconditional eBPF support in our packages.

You should be able to get it working by just removing the following line from the `debian/rules` file (should be line 39 if you have the most recent sources):

    packaging/bundle-libbpf.sh .

**Comment by mickae1** [comment]
Thx it works,

But because I've two version of openssl, the link failed.

How can I specify a version of openssl ?

example:
LIBSSL_CFLAGS = -I/usr/include
LIBSSL_LIBS = -L/usr/lib/arm-linux-gnueabihf -lssl

but I'm using this command:
dpkg-buildpackage -us -uc -rfakeroot

**Comment by mickae1** [comment]
I had the same issue in the past, but I could not find a solution ......

#7016
https://github.com/netdata/netdata/issues/7016

**Comment by mickae1** [comment]
and the final 
https://github.com/netdata/netdata/issues/8306

But Is there a way to force the version ?

I can make some modification somewhere if Have to ....

example:
LIBSSL_CFLAGS = -I/usr/include
LIBSSL_LIBS = -L/usr/lib/arm-linux-gnueabihf -lssl

**Comment by mickae1** [comment]
@Ferroin  any idea ? :'(

**Comment by Ferroin** [comment]
Sorry about the delayed response.

There’s currently not any reliable way to specify a particular OpenSSL version unless you’re trying to use a custom build that is not part of the main system libraries (and isn’t in the standard system paths), in which case ensuring that all of `LDFLAGS`, `CFLAGS`, `PKG_CONFIG_PATH` and `CMAKE_FLAGS` are set correctly for the desired copy of OpenSSL should work.

Patt of what complicates this is that our official package builds statically link a locally built copy of libwebsockets, and _that_ needs to be linked against the same version of OpenSSL as the main Netdata build is using. You _should_ be able to set this up by setting an appropriate `CMAKE_FLAGS` variable (see https://github.com/warmcat/libwebsockets/blob/main/READMEs/README.build.md for info on this, we use the standard build mechanisms for LWS, so the info in the upstream documentation is still accurate).

**Comment by vkalintiris** [comment]
@Ferroin Ok to close this now that we have packages for arm targets?

**Comment by mickae1** [comment]
> This should be fully resolved at this point. The regular kickstart installer should pull in a static build on Debian 8 armhf systems, which should work out of the box without issue.

Thanks, do you know where I can find the package in offline mode ?

**Comment by mickae1** [comment]
> This should be fully resolved at this point. The regular kickstart installer should pull in a static build on Debian 8 armhf systems, which should work out of the box without issue.

Thanks, do you know where I can find the package in offline mode ?

**Comment by mickae1** [comment]
@Ferroin  please :)
